File: spring-aws-s3-java/src/main/java/com/github/migangqui/spring/aws/s3/bean/UploadFileResult.java
package com.github.migangqui.spring.aws.s3.bean;

import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Builder
public class UploadFileResult {
	
	private String fileName;
	private int status;
	private String cause;
	private Exception exception;

}

File: spring-aws-s3-kotlin/src/main/kotlin/com/migangqui/spring/aws/s3/bean/UploadFileResult.kt
package com.migangqui.spring.aws.s3.bean

data class UploadFileResult(private val fileName: String? = null,
                            private val status: Int = 0,
                            private val cause: String? = null,
                            private val exception: Exception? = null) {
    constructor(fileName: String?, status: Int) : this(fileName, status, null, null)
}